base/sparse/linalg.jl

```diff
@@ -532,10 +531,13 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = 2)
     end
     ind = Array(Int64, n)
     ind_hist = Array(Int64, maxiter * t)
-    S = zeros(T, n, t)
+
+    Ti = T <: Real ? Float64 : Complex128
```

Comment:
It looks OK to me. Not sure we need to check whether `T` is a real type any more, though. I guess that `Ti = typeof(float(zero(T)))` would give the same `Ti` for complex and real types, which are the only ones we are interested in. A part from that, the PR seems ready to be merged.